### PR TITLE
Apply CRDs when cf-operator starts

### PR DESCRIPTION
**Pull request by edwardstudy**
[#167425313](https://www.pivotaltracker.com/story/show/167425313)

### CODE REVIEW

File: pkg/kube/apis/extendedjob/v1alpha1/register.go [hunk 1]
```diff
@@ -10,16 +13,116 @@ import (
...
+											Raw: []byte(`"manual"`),
+										},
+										{
+											Raw: []byte(`"once"`),
+										},
+										{
+											Raw: []byte(`"now"`),
+										},
+										{
+											Raw: []byte(`"podstate"`),
```
Comment (manno) [inline pkg/kube/apis/extendedjob/v1alpha1/register.go]: We no longer support podstate, should have been deleted when trigger ejobs were removed.

File: pkg/kube/operator/operator.go [hunk 2]
```diff
@@ -46,3 +51,75 @@ func NewManager(ctx context.Context, config *config.Config, cfg *rest.Config, op
...
 	return mgr, nil
 }
+
+// ApplyCRDs applies a collection of CRDs into the cluster
+func ApplyCRDs(config *rest.Config) error {
+	exClient, err := extv1client.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "Could not get kube client")
+	}
+	err = crd.ApplyCRD(
```
Comment (manno) [inline pkg/kube/operator/operator.go]: This repetition looks like we could put the values in a `[]resource{{name: ..., kind: ...}, {}}` and loop over that, but then it's probably not less code. But if `ApplyCRD` or `WaitForCRDReady` ever change, we only have to fix one invocation.
Well, I have no strong preference here.

Comment (edwardstudy) [reply pkg/kube/operator/operator.go]: It's better

File: cmd/internal/root.go [hunk 3]
```diff
@@ -129,6 +138,7 @@ func init() {
 	pf.Int("max-extendedjob-workers", 1, "Maximum of number concurrently running ExtendedJob controller")
 	pf.Int("max-extendedsecret-workers", 5, "Maximum of number concurrently running ExtendedSecret controller")
 	pf.Int("max-extendedstatefulset-workers", 1, "Maximum of number concurrently running ExtendedStatefulSet controller")
+	pf.Bool("apply-crd", true, "If true, apply CRDs on start")
```
Comment (manno) [inline cmd/internal/root.go]: false so we are backwards compatible for now

File: pkg/kube/apis/extendedjob/v1alpha1/register.go (hunk 1 above)
Comment (edwardstudy) [reply pkg/kube/apis/extendedjob/v1alpha1/register.go]: done

File: pkg/kube/operator/operator.go (hunk 2 above)
Comment (edwardstudy) [reply pkg/kube/operator/operator.go]: done

File: cmd/internal/root.go (hunk 3 above)
Comment (edwardstudy) [reply cmd/internal/root.go]: done

**Pull request merged**